cmp/grid/GridModel.js

```diff
@@ -692,12 +693,26 @@ export class GridModel {
      * Autosize columns to fit their contents.
      * @param {string|string[]} [colIds] - which columns to autosize; defaults to all leaf columns.
      */
-    autoSizeColumns(colIds = this.getLeafColumns().map(col => col.colId)) {
+    autoSizeColumns(colIds) {
+        if (!colIds) colIds = this.getLeafColumns().map(col => col.colId);
+
         colIds = castArray(colIds).filter(id => {
             const col = this.getColumn(id);
-            return col && !col.flex;
+            return col && col.resizable && !col.hidden && !col.flex;
         });
-        if (colIds.length) this.agColumnApi.autoSizeColumns(colIds);
+        if (colIds.length) {
+            start().then(async () => {
+                this.agApi.showLoadingOverlay();
+
+                const colStateChanges = XH.gridAutosizeService.autoSizeColumns({gridModel: this, colIds});
+                this.applyColumnStateChanges(colStateChanges);
+
+                console.debug('Columns autosized:', colStateChanges);
+
+                await wait(100);
```

Comment:
Could we comment this await? Is it here to allow the grid to complete its render/paint before removing the mask, or something else?